kubetest/kind/kind.go

```diff
@@ -121,33 +154,95 @@ func (k *Kind) setKubeConfigEnv() error {
 	return nil
 }
 
-// Up builds kind and the node image and then deploys a cluster based on the kind config.
-func (k *Kind) Up() error {
-	// TODO(neolit123): fetch a stable release binary and stash it in a kubetest specific dir.
-	// This should be the default option instead of building kind from source.
+// prepareKindBinary either builds kind from source or pulls a binary from GitHub.
+func (k *Kind) prepareKindBinary() error {
```

Comment:
TODO(bentheelder): have a set of hashes from [GOOS,GOARCH] -> hash and avoid re-downloading based on hash

Reply:
maybe we should write the binary as `kind-os-arch` instead of always as `kind`.